src/App.js

```diff
@@ -1,49 +1,71 @@
-import * as React from "react";
-import { Fragment } from "react";
-import { useState } from 'react';
-import { randomQuote } from "./randomizer.js";
+import React, { useState, useEffect } from 'react';
+import { Quote } from "./Quote.js";
+import { colorizer } from "./colorizer.js";
 
-function Quote({content, author}) {
-  return(
-    <Fragment>
-      <div id="quote">
-        <p id="text">{content}</p>
-        <p id="author">{"—" + author}</p>
-      </div>
-      <span id="tweet-container">
-        <img 
-          id="twitter-logo"
-          src="images/twitter.svg" 
-          alt="Twitter logo" 
-        />
-        <a 
-          id="tweet-quote" 
-          href={"https://twitter.com/intent/tweet?text=“" + content + "” —"+author + ",&via=iggldee"} 
-          target="_blank" 
-          rel="noopener noreferrer" 
-          OnClick="window.open(this.href)" 
-          title="Share on Twitter"
-        >
-          Tweet it<i>!</i>
-        </a>
-      </span>
-    </Fragment>
-  );
-}
+export default function Wrapper() {
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(null);
+  const [customError, setCustomError] = useState(null); // allow for custom-set, more explanatory error messages
+  const [loading, setLoading] = useState(true);
+  const [newQuote, setNewQuote] = useState(true); // establish this var so it can be called later and updated, in turn triggering useEffect's fetch() method to run again upon button click
+  const fetchOptions = {
+    method: 'GET',
+    headers: {
+      'X-RapidAPI-Key': '929124106emshf8e29cf6d861bf0p179527jsn25f292f15772',
+      'X-RapidAPI-Host': 'quotes15.p.rapidapi.com'
+    }
+  };
 
-function New({handleClick}) {
-  return <button id="new-quote" onClick={handleClick}>New quote</button>;
-}
+  // pass quote's length to colorizer() to determine color to associate it to in UI; used by both Quote and New components
+  let color;
+  data && (color = `lab(
+    85% 
+    ${colorizer(data.content.length).a} 
+    ${colorizer(data.content.length).b}
+    )`);
+
+  useEffect(() => {
+    setData(null);
```

Comment:
This useEffect() function is called upon both intial app load, and upon click of "New quote" button. Due to the latter, `setData()`, `setError()`, and `setLoading()` need to be set to their initial states again _within_ the function:
If the button is clicked when there had been an initial quote displayed, that quote needs to be removed (otherwise the loading state will display on top of the initial quote), hence `setData(null)`.
If the button is clicked when there had been an error displayed, that previous error needs to be reset (hence `setError(null)`). 
Also, the loading state needs to be reset to its initial state (`setLoading(true)`) because this state had been reset in the course of the previous use of useEffect (before the current button click) to `null`.